clusterloader2/pkg/measurement/common/probes/probes.go

```diff
@@ -0,0 +1,338 @@
+	manifests, err := filepath.Glob(expandedManifestGlob)
+	if err != nil {
+		return err
+	}
+	for _, manifest := range manifests {
+		klog.Infof("Applying %s\n", manifest)
+		obj, err := templateProvider.TemplateToObject(filepath.Base(manifest), p.templateMapping)
+		if err != nil {
+			return err
+		}
+		if err := p.framework.CreateObject(probesNamespace, obj.GetName(), obj); err != nil {
+			return fmt.Errorf("error while applying (%s): %v", manifest, err)
+		}
+		if obj.GetKind() == "ServiceMonitor" {
+			p.nServiceMonitors++
+		}
+	}
+	return nil
+}
+
+func (p *probesMeasurement) waitTillProbesReady() error {
+	klog.Info("Waiting for Probes to become ready...")
+	return wait.Poll(checkProbesReadyInterval, checkProbesReadyTimeout, p.checkProbesReady)
+}
+
+func (p *probesMeasurement) checkProbesReady() (bool, error) {
+	expectedTargets := p.replicasPerProbe * p.nServiceMonitors
+	return prometheus.CheckTargetsReady(
+		p.framework.GetClientSets().GetClient(), isProbeTarget, expectedTargets)
+}
+
+func isProbeTarget(t prometheus.Target) bool {
+	return t.Labels["namespace"] == probesNamespace
+}
+
+func parseThresholds(params map[string]interface{}) (map[string]*measurementutil.LatencyMetric, error) {
+	thresholds := make(map[string]*measurementutil.LatencyMetric)
+	for name, thresholdVal := range params["thresholds"].(map[string]interface{}) {
+		threshold, err := time.ParseDuration(thresholdVal.(string))
+		if err != nil {
+			return nil, err
+		}
+		thresholds[name] = makeLatencyThreshold(threshold)
+	}
+	return thresholds, nil
+}
+
+func makeLatencyThreshold(threshold time.Duration) *measurementutil.LatencyMetric {
+	return &measurementutil.LatencyMetric{
+		Perc50: threshold,
+		Perc90: threshold,
+		Perc99: threshold,
+	}
+}
+
+func prepareQuery(queryTemplate string, startTime, endTime time.Time) string {
+	measurementDuration := endTime.Sub(startTime)
+	return fmt.Sprintf(queryTemplate, measurementutil.ToPrometheusTime(measurementDuration))
+}
+
+type probesSummary struct {
+	probeSummaries []*probeSummary
+}
+
+type probeSummary struct {
+	name    string
+	latency measurementutil.LatencyMetric
+}
+
+// SummaryName returns name of the summary.
+func (p *probesSummary) SummaryName() string {
+	return name
+}
+
+// PrintSummary returns summary as a string.
+func (p *probesSummary) PrintSummary() (string, error) {
+	perfData := &measurementutil.PerfData{Version: currentProbesMetricsVersion}
+	for _, probe := range p.probeSummaries {
+		perfData.DataItems = append(perfData.DataItems, measurementutil.DataItem{
+			Data: map[string]float64{
+				"Perc50": float64(probe.latency.Perc50) / 1000000, // ns -> ms
+				"Perc90": float64(probe.latency.Perc90) / 1000000,
+				"Perc99": float64(probe.latency.Perc99) / 1000000,
+			},
+			Unit:   "ms",
+			Labels: map[string]string{"Metric": name},
+		})
+	}
+	return util.PrettyPrintJSON(perfData)
+}
+
+// TODO(mm4tt): Remove the method below and start using the one from common util once it's available.
```

Comment:
What is this blocked on? Can we do that now?

Reply:
What is the status of it?

Reply:
@krzysied, any updates?

Reply:
I haven't touch it yet.
Assuming that for some reason it is blocking I will look into this today.

Reply:
https://github.com/kubernetes/perf-tests/pull/502

Reply:
Thanks, will wait till https://github.com/kubernetes/perf-tests/pull/502 gets merged and remove this code.

Reply:
Done.